src/datasources/asset/AssetDataSource.ts

```diff
@@ -0,0 +1,325 @@
+      result.fields = [
+        { name: 'time', values: utilization_array_with_alias[0].datetimes },
+      ];
+      utilization_array_with_alias.forEach((data) => {
+        result.fields.push(
+          {
+            name: data.id,
+            values: data.values,
+            // default configuration for visualizations
+            config: {
+              displayName: `${data.alias} ${suffix}`,
+              unit: '%',
+              min: 0,
+              max: 100
+            }
+          }
+        )
+      })
+      return result;
+    }
+  }
+
+  shouldRunQuery(_: AssetQuery): boolean {
+    return true;
+  }
+
+  private async fetchAndProcessUtilizationData(query: AssetQuery, options: DataQueryRequest): Promise<Array<{
+    id: string,
+    datetimes: number[],
+    values: number[]
+  }>> {
+    const [from, to] = [options.range.from.valueOf(), options.range.to.valueOf()];
+    const workspace_id = this.templateSrv.replace(query.workspace);
+    const minion_id = this.templateSrv.replace(query.minionId);
+    // this value is set in the Datasource -> Query options -> Max data points
+    const maxDataPoints = options.maxDataPoints;
+    let continuationToken = '';
+    const {
+      assetIdentifier,
+      minionId,
+      utilizationCategory,
+      isPeak,
+      peakDays,
+      timeFrequency
+    } = query;
+    const workingHoursPolicy = {
+      'startTime': momentToTime(dateTime(query.peakStart)),
+      'endTime': momentToTime(dateTime(query.nonPeakStart))
+    }
+    const utilizationFilter: string = utilizationCategory === UtilizationCategory.TEST ? `Category = \"Test\" or Category = \"test\"` : '';
+    let arrayOfEntityIds: string[];
+    if (query.entityType === EntityType.ASSET) {
+      arrayOfEntityIds = replaceVariables([assetIdentifier], this.templateSrv);
+    } else {
+      arrayOfEntityIds = replaceVariables([minionId], this.templateSrv);
+    }
+    // filter empty values
+    arrayOfEntityIds = arrayOfEntityIds.filter(Boolean);
+    const resultArray: Array<{
+      id: string,
+      datetimes: number[],
+      values: number[]
+    }> = [];
+    if (!arrayOfEntityIds.length) {
+      if (query.entityType === EntityType.ASSET) {
+        let conditions: string[] = [];
+        if (workspace_id) {
+          conditions.push(`workspace = "${workspace_id}"`);
+        }
+        if (minion_id) {
+          conditions.push(`${AssetFilterProperties.LocationMinionId} = "${minion_id}"`);
+        }
+        const assetFilter = conditions.join(' and ');
+        const assetsResponse = await this.queryAssets(assetFilter, 25);
+        arrayOfEntityIds = assetsResponse.map((asset) => {
+          return asset.id;
+        })
+      } else {
+        let filterString = '';
+        if (query.workspace) {
+          filterString += `workspace = "${query.workspace}"`;
+        }
+        let systemMetadata = await this.post<{ data: SystemMetadata[] }>(this.sysmgmtUrl + '/query-systems', {
```

Comment:
This should use the shared function for querying systems—add more params to the function if needed.